docs/source/usage.rst

```diff
@@ -12,7 +15,206 @@ To use STAC Task, first install it using pip:
+                           [101.0, 0.0],
+                           [101.0, 1.0],
+                           [100.0, 1.0],
+                           [100.0, 0.0],
+                     ]
+                  ],
+               },
+               bbox=None,
+               datetime=datetime.now(timezone.utc),
+               properties={},
+         )
+
+         return [item.to_dict()]
+
+   # Support for running as a Lambda Function
+   def handler(event: dict[str, Any], context: Any) -> dict[str, Any]:
+      return MyTask.handler(event)  # type: ignore
+
+   # Support for running as a CLI application
+   if __name__ == "__main__":
+      MyTask.cli()
+
+The expected input looks like the following:
+
+.. code-block:: json
+
+   {
+      "id": "my-task/workflow-my-task/5427299e5b635537f33c07e0ad32fb87",
+      "item_id": "G23923",
+      "process": {
+         "upload_options": {
+            "collections": {
+               "my-collection": "$[?(@.id =~ '.*')]"
+            }
+         }
+      }
+   }
+
+In Task, the `/process/upload_options/collections` mapping uses JSONPath to map attributes of the
+output Item to the collection that should be assigned to it. In this case, we only have one defined
+that matches on any `id` value, and sets the collection to `my-collection`.
+
+Running this with `python my-task.py run --local in.json` results in the following
+output JSON, which has modified the payload to add a new
+`features` attribute array.
+
+.. code-block:: json
+
+   {
+      "id": "my-task/workflow-my-task/5427299e5b635537f33c07e0ad32fb87",
+      "item_id": "G23923",
+      "process": {
+         "upload_options": {
+            "collections": {
+            "my-collection": "$[?(@.id =~ '.*')]"
+            }
+         }
+      },
+      "features": [
+         {
+            "type": "Feature",
+            "stac_version": "1.0.0",
+            "id": "G23923",
+            "properties": {
+            "datetime": "2024-04-04T13:55:05.598886Z",
+            "processing:software": {
+               "my-task": "v2024.02.01"
+            }
+            },
+            "geometry": {
+            "type": "Polygon",
+            "coordinates": [
+               [
+                  [ 100.0, 0.0 ],
+                  [ 101.0, 0.0 ],
+                  [ 101.0, 1.0 ],
+                  [ 100.0, 1.0 ],
+                  [ 100.0, 0.0 ]
+               ]
+            ]
+            },
+            "links": [],
+            "assets": {},
+            "stac_extensions": [
+            "https://stac-extensions.github.io/processing/v1.1.0/schema.json"
+            ],
+            "collection": "my-collection"
+         }
+      ]
+   }
+
+CLI Usage
+---------
+
+To run a Task as a CLI application, add a main definition to the class inheriting Task:
+
+.. code-block:: python
+
+   if __name__ == "__main__":
+      MyTask.cli()
+
+This provides a CLI that supports several useful flags for using stactask. Invoking it
```

Comment:
`stactask` -> `stac-task`

Reply:
I made them consistent -- stac-task for the general library name, stactask for the package, STAC Task for the formal name.